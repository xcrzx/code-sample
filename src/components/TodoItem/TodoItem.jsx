import React, { PropTypes } from 'react'
import classnames from 'classnames'

import './TodoItem.scss'

const TodoItem = ({ title, completed, ...rest }) => (
  <div {...rest} className={classnames('todo-item', completed && 'completed')}>
    {title}
  </div>
)

TodoItem.propTypes = {
  title: PropTypes.string,
  completed: PropTypes.bool,
}

export default TodoItem
