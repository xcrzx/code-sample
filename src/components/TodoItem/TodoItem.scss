.todo-item {
  padding: 20px;
  cursor: pointer;

  + .todo-item {
    border-top: 1px solid #e1e1e1;
  }

  &.completed {
    text-decoration: line-through;
  }
}