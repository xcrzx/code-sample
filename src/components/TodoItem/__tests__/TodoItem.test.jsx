import React from 'react'
import { shallow } from 'enzyme'

import TodoItem from '../TodoItem'

describe('TodoItem component', () => {
  test('default render', () => {
    const component = shallow(<TodoItem />)
    expect(component).toMatchSnapshot()
  })

  test('with title', () => {
    const component = shallow(<TodoItem title="Title" />)
    expect(component).toMatchSnapshot()
  })

  test('completed', () => {
    const component = shallow(<TodoItem completed />)
    expect(component).toMatchSnapshot()
  })

  test('onClick', () => {
    const clickHandler = jest.fn()

    const component = shallow(<TodoItem id={1} onClick={clickHandler} />)
    component.simulate('click')

    expect(clickHandler).toHaveBeenCalledWith(1)
  })
})
