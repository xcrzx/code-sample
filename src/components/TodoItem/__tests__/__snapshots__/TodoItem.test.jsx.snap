// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem component completed 1`] = `
<div
  className="todo-item completed"
  onClick={[Function]}
/>
`;

exports[`TodoItem component default render 1`] = `
<div
  className="todo-item"
  onClick={[Function]}
/>
`;

exports[`TodoItem component with title 1`] = `
<div
  className="todo-item"
  onClick={[Function]}
>
  Title
</div>
`;
